lib/bitex_bot/models/api_wrappers/bitfinex/bitfinex_api_wrapper.rb

```diff
@@ -1,119 +1,129 @@
-  end
+    # {
+    #   id: 448411365, symbol: 'btcusd', exchange: 'bitfinex', price: '0.02', avg_execution_price: '0.0',  side: 'buy',
+    #   type: 'exchange limit', timestamp: '1444276597.0', is_live: true, is_cancelled: false, is_hidden: false,
+    #   was_forced: false, original_amount: '0.02', remaining_amount: '0.02', executed_amount: '0.0'
+    # }
+    def order_parser(order)
+      self::Order
+        .new(order[:id].to_s, order[:side].to_sym, order[:price].to_d, order[:original_amount].to_d, order[:timestamp].to_i)
+    end
 
-  # [
-  #   { type: 'deposit', currency: 'btc', amount: '0.0', available: '0.0' },
-  #   { type: 'deposit', currency: 'usd', amount: '1.0', available: '1.0' },
-  #   { type: 'exchange', currency: 'btc', amount: '1', available: '1' }
-  # ]
-  def self.balance_summary_parser(b)
-    BalanceSummary.new.tap do |summary|
-      btc = b.find { |balance| balance[:currency] == 'btc' } || {}
-      summary[:btc] = Balance.new(btc[:amount].to_d, btc[:amount].to_d - btc[:available].to_d, btc[:available].to_d)
+    # [
+    #   { type: 'deposit', currency: 'btc', amount: '0.0', available: '0.0' },
+    #   { type: 'deposit', currency: 'usd', amount: '1.0', available: '1.0' },
+    #   { type: 'exchange', currency: 'btc', amount: '1', available: '1' }
+    # ]
+    def balance_summary_parser(balances)
+      self::BalanceSummary.new(
+        balance_parser(balances, 'btc'),
+        balance_parser(balances, 'usd'),
+        client.account_info.first[:taker_fees].to_d
+      )
+    end
 
-      usd = b.find { |balance| balance[:currency] == 'usd' } || {}
-      summary[:usd] = Balance.new(usd[:amount].to_d, usd[:amount].to_d - usd[:available].to_d, usd[:available].to_d)
+    def balance_parser(balances, currency)
+      currency_balance = balances.find { |balance| balance[:currency] == currency } || {}
+      self::Balance.new(
+        currency_balance[:amount].to_d,
+        currency_balance[:amount].to_d - currency_balance[:available].to_d,
+        currency_balance[:available].to_d
+      )
+    end
 
-      summary[:fee] = client.account_info.first[:taker_fees].to_d
+    # {
+    #   bids: [{ price: '574.61', amount: '0.14397', timestamp: '1472506127.0' }],
+    #   asks: [{ price: '574.62', amount: '19.1334', timestamp: '1472506126.0 '}]
+    # }
+    def order_book_parser(book)
+      self::OrderBook.new(Time.now.to_i, order_summary(book[:bids]), order_summary(book[:asks]))
     end
-  end
 
-  # {
-  #   bids: [{ price: '574.61', amount: '0.14397', timestamp: '1472506127.0' }],
-  #   asks: [{ price: '574.62', amount: '19.1334', timestamp: '1472506126.0 '}]
-  # }
-  def self.order_book_parser(b)
-    OrderBook.new(
-      Time.now.to_i,
-      b[:bids].map { |bid| OrderSummary.new(bid[:price].to_d, bid[:amount].to_d) },
-      b[:asks].map { |ask| OrderSummary.new(ask[:price].to_d, ask[:amount].to_d) }
-    )
+    def order_summary(stock_market)
```

Comment:
`stock_market` is actually an array of orders, I think it looks better if we just call them `orders`.

Reply:
Roger!